### PR TITLE
Safer URL parsing. Disallow empty container name.

### PR DESCRIPTION
**Pull request by gtopper**
Various fixes (see commit log).

All tests pass.

### CODE REVIEW

File: pkg/dataplane/http/context.go [hunk 1]
```diff
@@ -20,6 +19,7 @@ import (
 	"github.com/v3io/v3io-go/pkg/errors"
 
 	"github.com/nuclio/logger"
+	"github.com/pkg/errors"
```
Comment (pavius) [inline pkg/dataplane/http/context.go]: We don't use pkg/errors - we use the nuclio/errors

File: pkg/errors/errors.go [hunk 2]
```diff
@@ -11,13 +11,13 @@ type ErrorWithStatusCode struct {
...
-	return &ErrorWithStatusCode{
+func NewErrorWithStatusCode(err error, statusCode int) ErrorWithStatusCode {
+	return ErrorWithStatusCode{
 		error:      err,
 		statusCode: statusCode,
 	}
 }
 
-func (e *ErrorWithStatusCode) StatusCode() int {
+func (e ErrorWithStatusCode) StatusCode() int {
```
Comment (pavius) [inline pkg/errors/errors.go]: Why not a pointer receiver? This'll copy the entire structure each time it's called

Comment (gtopper) [reply pkg/errors/errors.go]: Use ErrorWithStatusCode by value to avoid breaking compatibility in a really bad way.

Comment (gtopper) [reply pkg/errors/errors.go]: Also, the structure contains just an int and an interface, so very little data.

**Pull request merged**